temporalio/worker/_tuning.py

```diff
@@ -0,0 +1,214 @@
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from datetime import timedelta
+from typing import Literal, Optional, Union
+
+from typing_extensions import TypeAlias
+
+import temporalio.bridge.worker
+
+_DEFAULT_RESOURCE_ACTIVITY_MAX = 500
+
+
+@dataclass(frozen=True)
+class FixedSizeSlotSupplier:
+    """A fixed-size slot supplier that will never issue more than a fixed number of slots."""
+
+    num_slots: int
+    """The maximum number of slots that can be issued"""
+
+
+@dataclass(frozen=True)
+class ResourceBasedTunerConfig:
+    """Options for a :py:class:`ResourceBasedTuner` or a :py:class:`ResourceBasedSlotSupplier`.
+
+    .. warning::
+        The resource based tuner is currently experimental.
+    """
+
+    target_memory_usage: float
+    """A value between 0 and 1 that represents the target (system) memory usage. It's not recommended
+       to set this higher than 0.8, since how much memory a workflow may use is not predictable, and
+       you don't want to encounter OOM errors."""
+    target_cpu_usage: float
+    """A value between 0 and 1 that represents the target (system) CPU usage. This can be set to 1.0
+       if desired, but it's recommended to leave some headroom for other processes."""
+
+
+@dataclass(frozen=True)
+class ResourceBasedSlotOptions:
```

Comment:
```suggestion
class ResourceBasedSlotConfig:
```

Let's change this to `Config` too